tests/php/test-fieldmanager-context-menuitem.php

```diff
@@ -0,0 +1,176 @@
+									array(
+										'limit'    => 0,
+										'name'     => 'extext',
+										'one_label_per_item' => false,
+										'sortable' => true,
+										'index'    => '_extext_index',
+									)
+								),
+							),
+						)
+					),
+				),
+			)
+		);
+	}
+
+	public function test_context_render() {
+		global $wp_version;
+
+		// Only run these tests for WP versions above 5.4.0.
+		if ( version_compare( $wp_version, '5.4.0', '<' ) ) {
+			return;
+		}
+
+		$base = $this->_get_elements();
+		ob_start();
+		$base->add_nav_menu_fields()->add_fields( $this->post->ID );
+		$str = ob_get_clean();
+		// we can't really care about the structure of the HTML, but we can make sure that all fields are here
+		$this->assertRegExp( '/<input[^>]+type="hidden"[^>]+name="fieldmanager-base_group_fm-menu-item-id-' . $this->post->ID . '-nonce"/', $str );
+		$this->assertRegExp( '/<input[^>]+type="text"[^>]+name="base_group_fm-menu-item-id-' . $this->post->ID . '\[test_basic\]"/', $str );
+		$this->assertRegExp( '/<input[^>]+type="text"[^>]+name="base_group_fm-menu-item-id-' . $this->post->ID . '\[test_textfield\]"/', $str );
+		$this->assertRegExp( '/<textarea[^>]+name="base_group_fm-menu-item-id-' . $this->post->ID . '\[test_htmlfield\]"/', $str );
+		$this->assertContains( 'name="base_group_fm-menu-item-id-' . $this->post->ID . '[test_extended][0][extext][proto]"', $str );
+		$this->assertContains( 'name="base_group_fm-menu-item-id-' . $this->post->ID . '[test_extended][0][extext][0]"', $str );
+	}
+
+	public function test_context_save() {
+		global $wp_version;
+
+		// Only run these tests for WP versions above 5.4.0.
+		if ( version_compare( $wp_version, '5.4.0', '<' ) ) {
+			return;
+		}
+
+		$base      = $this->_get_elements();
+		$test_data = $this->_get_valid_test_data();
+
+		$base->add_nav_menu_fields()->save_to_post_meta( $this->post_id, $test_data['base_group'] );
+
+		$saved_value = get_post_meta( $this->post_id, 'base_group', true );
+		$saved_index = get_post_meta( $this->post_id, '_test_index', true );
+
+		$this->assertEquals( $saved_value['test_basic'], 'lorem ipsum' );
+		$this->assertEquals( $saved_index, $saved_value['test_textfield'] );
+		$this->assertEquals( $saved_value['test_textfield'], 'alley interactive' );
+		$this->assertEquals( $saved_value['test_htmlfield'], '<b>Hello</b> world' );
+		$this->assertEquals( count( $saved_value['test_extended'] ), 4 );
+		$this->assertEquals( count( $saved_value['test_extended'][0]['extext'] ), 1 );
+		$this->assertEquals( count( $saved_value['test_extended'][1]['extext'] ), 3 );
+		$this->assertEquals( count( $saved_value['test_extended'][2]['extext'] ), 1 );
+		$this->assertEquals( count( $saved_value['test_extended'][3]['extext'] ), 1 );
+		$this->assertEquals( $saved_value['test_extended'][1]['extext'], array( 'second1', 'second2', 'second3' ) );
+		$this->assertEquals( $saved_value['test_extended'][3]['extext'][0], 'fourth' );
+	}
+
+	public function test_programmatic_save_posts() {
```

Comment:
This test doesn't contain any assertions.

Reply:
Good catch :) I was just following this code, but I did not vet it out entirely - https://github.com/alleyinteractive/wordpress-fieldmanager/blob/1a402804ba5d89d524a28b44fbd1e3f13463d63a/tests/php/test-fieldmanager-context-post.php#L135